thirdparty/runtime_environments.go

```diff
@@ -308,3 +312,101 @@ func (c *RuntimeEnvironmentsClient) getHistory(ctx context.Context, opts DistroH
 	}
 	return amiHistory, nil
 }
+
+// ImageEventEntry represents a change to the image.
+type ImageEventEntry struct {
+	Name   string
+	After  string
+	Before string
+	Type   string
+	Action string
+}
+
+// ImageEvent contains information about changes to an image when the ami changes.
+type ImageEvent struct {
+	Entries   []ImageEventEntry
+	Timestamp time.Time
+	AMIBefore string
+	AMIAfter  string
+}
+
+// EventHistoryOptions represents the filtering arguments for getEvents. Image is a required argument.
+type EventHistoryOptions struct {
+	Image string
+	Page  int
+	Limit int
+}
+
+// stringToTime converts a string representing time to type time.Time.
+func stringToTime(timeInitial string) (time.Time, error) {
+	timestamp, err := strconv.ParseInt(timeInitial, 10, 64)
+	if err != nil {
+		return time.Time{}, errors.Wrapf(err, "converting time: '%s'", timeInitial)
+	}
+	return time.Unix(timestamp, 0), nil
+}
+
+// buildImageEventEntry make an ImageEventEntry given an ImageDiffChange.
+func buildImageEventEntry(diff ImageDiffChange) *ImageEventEntry {
+	action := ""
+	if diff.Added != "" && diff.Removed != "" {
+		action = UpdatedImageEntryAction
+	} else if diff.Added != "" {
+		action = AddedImageEntryAction
+	} else if diff.Removed != "" {
+		action = DeletedImageEntryAction
+	}
+	entry := ImageEventEntry{
+		Name:   diff.Name,
+		After:  diff.Added,
+		Before: diff.Removed,
+		Type:   diff.Type,
+		Action: action,
+	}
+	return &entry
+}
+
+// getEvents returns information about the changes between AMIs that occurred on the image.
+func (c *RuntimeEnvironmentsClient) getEvents(ctx context.Context, opts EventHistoryOptions) ([]ImageEvent, error) {
+	if opts.Limit == 0 {
+		opts.Limit = 9
+	}
+	optsHistory := DistroHistoryFilterOptions{
+		Distro: opts.Image,
+		Page:   opts.Page,
+		Limit:  opts.Limit + 1,
+	}
+	imageHistory, err := c.getHistory(ctx, optsHistory)
+	if err != nil {
+		return nil, errors.Wrap(err, "getting image history")
+	}
+	result := []ImageEvent{}
+	for i := 0; i < len(imageHistory)-1; i++ {
```

Comment:
would this be easier to read if you just start from i:=1 so you don't have to do +1 or -1? 

if you do, you should say something in a comment so it's apparent

Reply:
In either case, I'll have to do +1 or -1 because I have to access the beforeAMI and the afterAMI which are off by 1 from each other.